packages/openapi-generator/src/sdk-metadata/code-sample-util.ts

```diff
@@ -0,0 +1,167 @@
+import levenstein from 'fast-levenshtein';
+import { codeBlock, pascalCase } from '@sap-cloud-sdk/util';
+import { OpenApiApi, OpenApiOperation } from '../openapi-types';
+
+const distanceThreshold = 5;
+export function getMainApi(
+  serviceName: string,
+  apis: OpenApiApi[]
+): OpenApiApi {
+  let closestApi: OpenApiApi | undefined;
+  let minDistance = distanceThreshold;
+
+  apis.forEach(api => {
+    const distance = getLevensteinDistance(serviceName, api.name);
+    if (distance <= minDistance) {
+      minDistance = distance;
+      closestApi = api;
+    }
+  });
+  if (closestApi) {
+    return closestApi;
+  }
+  // If no closest api found, return the api with most operations
+  const sortedByOperationsLength = apis.sort((a, b) =>
+    a.operations.length > b.operations.length ? -1 : 1
+  );
+  const withGetAll = sortedByOperationsLength.filter(api =>
+    api.operations.find(operation =>
+      operation.operationId.toLocaleLowerCase('getall')
+    )
+  );
+  if (withGetAll) {
+    return withGetAll[0];
+  }
+  // return the one with most methods
+  return sortedByOperationsLength[0];
+}
+
+export function getMainOperation(api: OpenApiApi): OpenApiOperation {
+  if (api.operations.length === 1) {
+    return api.operations[0];
+  }
+  return (
+    getLevensteinClosestOperation(api.name, api.operations) ||
+    getGetAllOperation(api.operations) ||
+    getGetOperation(api.operations) ||
+    getAnyOperationWithoutParams(api.operations) ||
+    api.operations[0]
+  );
+}
+
+export function getLevensteinClosestOperation(
```

Comment:
Perhaps you can make a reuse method so that you can use this in the case above. Potentential signaure:
getLevensteinClosestOperation<T>(name:string,objectToCheck:T,extractorFn:(T)=>string,options?:{threshold:number}):T[]|undefined